### ISSUE TITLE
flush API cache / force redownload of API

### ISSUE THREAD

**Issue by jeffeaton** [OJWatson/rdhs#23]
We got a problem. I have cached version of an API call. But I want to query the API again for updated results. 

1) API calls should have an argument force re-querying the API.
2) It would be convenient to have a function to flush the entire API cache.

**Comment by OJWatson** [comment]
Every time you start a new r session you create a new client in the back end without knowing it. This will then check for any updates to the api since the last time you used rdhs. If there is any update then the API calls are flushed.

So presuming there is not a new update while you are within your rsession this should never be an issue. However, for I suppose the issue where it does update while you are in an r session then a request forcing the requery could be useful. Will put one in for this (theres one in the get_available_datasets for this sort of thing where you've requested new datasets) and one to flush API calls.

I wonder though how useful it will be as how will someone know or come to find out that during an r session there has been an API update? I only make the check for updates at the beginning of a session so you don't query the data updates endpoint on every API request (to save time/overhead on their server). If you have any ideas of how it makes sense to alert the user about this let me know. Something more intelligent than my first thought of every x minutes since your session started do a check.



From: Jeff Eaton
Sent: Wednesday, May 2, 11:09 PM
Subject: [OJWatson/rdhs] flush API cache / force redownload of API (#23)
To: OJWatson/rdhs
Cc: Subscribed


We got a problem. I have cached version of an API call. But I want to query the API again for updated results.
API calls should have an argument force re-querying the API. It would be convenient to have a function to flush the entire API cache.

**Comment by jeffeaton** [comment]
Thanks -- I think that a check every time you start a new session should be more than sufficient, that sounds good to me.

The issue that arose yesterday was that when I tried to confirm the change to the DR2007SPE surveyId, it did not requery the API, and I had to manually go into my rdhs root and clear out the API cache folder. I thought that I had started a new session for that, but perhaps I didn’t.


On 3 May 2018, at 09:16, OJ Watson <notifications@github.com<mailto:notifications@github.com>> wrote:

Every time you start a new r session you create a new client in the back end without knowing it. This will then check for any updates to the api since the last time you used rdhs. If there is any update then the API calls are flushed.

So presuming there is not a new update while you are within your rsession this should never be an issue. However, for I suppose the issue where it does update while you are in an r session then a request forcing the requery could be useful. Will put one in for this (theres one in the get_available_datasets for this sort of thing where you've requested new datasets) and one to flush API calls.

I wonder though how useful it will be as how will someone know or come to find out that during an r session there has been an API update? I only make the check for updates at the beginning of a session so you don't query the data updates endpoint on every API request (to save time/overhead on their server). If you have any ideas of how it makes sense to alert the user about this let me know. Something more intelligent than my first thought of every x minutes since your session started do a check.



From: Jeff Eaton
Sent: Wednesday, May 2, 11:09 PM
Subject: [OJWatson/rdhs] flush API cache / force redownload of API (#23)
To: OJWatson/rdhs
Cc: Subscribed


We got a problem. I have cached version of an API call. But I want to query the API again for updated results.
API calls should have an argument force re-querying the API. It would be convenient to have a function to flush the entire API cache.
